cmd/controller-manager/main.go

```diff
@@ -192,3 +208,26 @@ func main() {
 	}
 	select {}
 }
+
+func parseControllers(controllers []string) (sets.Set[string], error) {
+	if len(controllers) == 0 {
+		return nil, nil
+	}
+
+	for _, name := range controllers {
+		if strings.ToLower(name) != name {
+			return nil, fmt.Errorf("upper case letters not allowed for controller name %q", name)
+		}
+	}
+
+	return sets.New(controllers...), nil
+}
```

Comment:
FYI, I do not think we need a whole func definition. Just use `sets.New` and `sets.IsSubset` directly in the `main` func.

Reply:
Killed `parseControllers` (and `shouldStartController` as well).